### PR TITLE
Update Persist.java

### PR DESCRIPTION
**Pull request by sh0inx**
updated load method to allow child plugin to recover from a bad config load

### CODE REVIEW

File: plugin/src/main/java/com/iridium/iridiumcore/Persist.java [hunk 1]
```diff
@@ -182,7 +184,19 @@ public <T> T load(Class<T> clazz, File file) {
                 return objectMapper.readValue(file, clazz);
             } catch (IOException e) {
                 javaPlugin.getLogger().severe("Failed to parse " + file + ": " + e.getMessage());
-                Bukkit.getPluginManager().disablePlugin(javaPlugin);
+                javaPlugin.getLogger().severe("Getting a backup for " + file + " into backups folder");
+                try {
+                    file.renameTo(new File(javaPlugin.getDataFolder(), "broken_" + file.getName() + persistType.getExtension()));
+                    File backupFolder = new File(javaPlugin.getDataFolder().getPath(), "backups");
+                    backupFolder.mkdirs();
+                    Files.move(file.toPath(), backupFolder.toPath(), StandardCopyOption.REPLACE_EXISTING);
```
Comment (PeachesMLG) [inline plugin/src/main/java/com/iridium/iridiumcore/Persist.java:192]: im confused, do we not need to disable the plugin after this? surely this method needs to return something?

Comment (sh0inx) [reply plugin/src/main/java/com/iridium/iridiumcore/Persist.java:192]: see 198, we only disable the plugin if it fails to make the backup (there are 2 try/catch, one is for loading the config, the other is for loading the default and moving the broken config to the backup folder)

Comment (PeachesMLG) [reply plugin/src/main/java/com/iridium/iridiumcore/Persist.java:192]: Yes, but what if it fails to load the origional, but then creates a backup successfully, what does this method return?

Comment (PeachesMLG) [reply plugin/src/main/java/com/iridium/iridiumcore/Persist.java:192]: ok, never mind Line 203, it will create a new instance

**Pull request merged**